ITensorTDVP.jl v0.4 (#74)
ITensor/ITensorTDVP.jl: README.md

## News

### ITensorTDVP.jl v0.4 Release Notes

#### Breaking changes

- When calling `tdvp(operator, t, init; kwargs...)`, `t` is now interpreted as the total evolution time, while in `ITensorTDVP.jl` v0.3 and below it was being interpreted as the time step. To upgrade, change code like:
```julia
tdvp(operator, -0.1im, init; nsweeps=10, kwargs...)
```
to:
```julia

> tdvp(operator, -1.0im, init; nsweeps=10, kwargs...)
> ```
> to evolve to time `1.0` over `10` steps with time steps of `0.1`. Also note that `ITensorTDVP.jl` v0.4 introduces `nsteps` as an alias for `nsweeps` in the `tdvp` function. `nsteps` is now the preferred syntax for specifying the number of time steps, and `nsweeps` may be deprecated in `tdvp` in the future. For example:

Author: ```suggestion
tdvp(operator, -1.0im, init; nsweeps=10, kwargs...)
```
to evolve to total time `1.0` over `10` time steps of size `0.1`. Also note that `ITensorTDVP.jl` v0.4 introduces `nsteps` as an alias for `nsweeps` in the `tdvp` function. `nsteps` is now the preferred syntax for specifying the number of time steps, and `nsweeps` may be deprecated in `tdvp` in the future. For example:
```

> to evolve to time `1.0` over `10` steps with time steps of `0.1`. Also note that `ITensorTDVP.jl` v0.4 introduces `nsteps` as an alias for `nsweeps` in the `tdvp` function. `nsteps` is now the preferred syntax for specifying the number of time steps, and `nsweeps` may be deprecated in `tdvp` in the future. For example:

Author: ```suggestion
to evolve to total time `1.0` over `10` time steps of size `0.1`. Also note that `ITensorTDVP.jl` v0.4 introduces `nsteps` as an alias for `nsweeps` in the `tdvp` function. `nsteps` is now the preferred syntax for specifying the number of time steps, and `nsweeps` may be deprecated in `tdvp` in the future. For example:
```

## News

### ITensorTDVP.jl v0.4 Release Notes

#### Breaking changes

- When calling `tdvp(operator, t, init; kwargs...)`, `t` is now interpreted as the total evolution time, while in `ITensorTDVP.jl` v0.3 and below it was being interpreted as the time step. To upgrade, change code like:
```julia
tdvp(operator, -0.1im, init; nsweeps=10, kwargs...)
```
to:
```julia

> t = -1.0im
> nsweeps = 10
> tdvp(operator, t, init; nsweeps, kwargs...)

Author: ```suggestion
tdvp(operator, -1.0im, init; nsweeps=10, kwargs...)
```

## News

### ITensorTDVP.jl v0.4 Release Notes

#### Breaking changes

> - When calling `tdvp(operator, t, init; kwargs...)`, `t` is now interpreted as the total evolution time, while in `ITensorTDVP.jl` it was being interpreted as the time step. To upgrade, change code like:

Author: ```suggestion
- When calling `tdvp(operator, t, init; kwargs...)`, `t` is now interpreted as the total evolution time, while in `ITensorTDVP.jl` v0.3 and below it was being interpreted as the time step. To upgrade, change code like:
```

```julia

> t = -0.1im
> nsweeps = 10
> tdvp(operator, t, init; nsweeps, kwargs...)

Author: ```suggestion
tdvp(operator, -0.1im, init; nsweeps=10, kwargs...)
```

```
to:
```julia
t = -1.0im
nsweeps = 10
tdvp(operator, t, init; nsweeps, kwargs...)
```

> to evolve to time `1.0` over `10` steps with time steps of `0.1`. Also note that `ITensorTDVP.jl` v0.4 instroduces `nsteps` as an alias for `nsweeps` (and the recommended syntax), so you can also use:

Author: ```suggestion
to evolve to time `1.0` over `10` steps with time steps of `0.1`. Also note that `ITensorTDVP.jl` v0.4 introduces `nsteps` as an alias for `nsweeps` in the `tdvp` function. `nsteps` is now the preferred syntax for specifying the number of time steps, and `nsweeps` may be deprecated in `tdvp` in the future. For example:
```

```
- Only the argument ordering `tdvp(operator, t, init; kwargs...)` is now supported. `tdvp(t, operator, init; kwargs...)` and `tdvp(operator, init, t; kwargs...)` have been removed.
- In `tdvp`, the keyword argument `solver_backend` has been renamed to `updater_backend`. Change code like:
```julia
tdvp(operator, t, init; solver_backend="applyexp", kwargs...)
```
to:
```julia
tdvp(operator, t, init; updater_backend="applyexp", kwargs...)
```
- In `tdvp` and `ITensorTDVP.dmrg`, keyword arguments passed to the local solver should now be passed in a NamedTuple in the `updater_kwargs` keyword argument, such as `updater_kwargs=(; tol=1e-5, krylovdim=20)`, instead of as keyword arguments `solver_tol`, `solver_krylovdim`, etc. Change code like:
```julia
tdvp(operator, t, init; solver_tol=1e-5, krylovdim=20, kwargs...)
ITensorTDVP.dmrg(operator, init; solver_tol=1e-5, solver_krylovdim=20, kwargs...)
```
to:
```julia
tdvp(operator, t, init; updater_kwargs=(; tol=1e-5, krylovdim=20), kwargs...)
ITensorTDVP.dmrg(operator, init; updater_kwargs=(; tol=1e-5, krylovdim=20), kwargs...)
```
- In `linsolve`, the keyword argument `solver_kwargs` has been renamed to `updater_kwargs`.

> - In `ITensorTDVP.dmrg`, `dmrg_x`, and `linsolve`, the keyword argument `step_observer!` has been renamed to `sweep_observer!`. Either name is allowed in `tdvp`.

Author: ```suggestion
- In `ITensorTDVP.dmrg`, `dmrg_x`, and `linsolve`, the keyword argument `step_observer!` has been renamed to `sweep_observer!`. Either name is allowed in `tdvp` but `step_observer!` is preferred and the name `sweep_observer!` may be deprecated in `tdvp` in future versions.
```

#### Breaking changes

> - ITensorTDVP.jl v0.2.0-v0.2.4: The `applyexp` Krylov exponentiation updater backend was removed, and `updater_backend="applyexp"` option for `tdvp` now just calls `exponentiate` from KrylovKit.jl. `applyexp` is in many ways the same as `exponentiate` bit `exponentiate` has more advanced features like restarts. In these versions, `updater_backend="applyexp"` prints a warning to that effect. As of ITensorTDVP.jl v0.2.5, we have brought back the `applyexp` backend because we received reports that it performed better in certain cases. We plan to investigate that issue and make sure `exponentiate` works as well as `applyexp` in those cases so that we can go back to just having a single `exponentiate` backend.

Author: ```suggestion
- ITensorTDVP.jl v0.2.0-v0.2.4: The `applyexp` Krylov exponentiation solver backend was removed, and `solver_backend="applyexp"` option for `tdvp` now just calls `exponentiate` from KrylovKit.jl. `applyexp` is in many ways the same as `exponentiate` bit `exponentiate` has more advanced features like restarts. In these versions, `solver_backend="applyexp"` prints a warning to that effect. As of ITensorTDVP.jl v0.2.5, we have brought back the `applyexp` backend because we received reports that it performed better in certain cases. We plan to investigate that issue and make sure `exponentiate` works as well as `applyexp` in those cases so that we can go back to just having a single `exponentiate` backend.
```
